src/RadixDlt.NetworkGateway.Abstractions/Utilities/ClampDateTimeFormatter.cs

```diff
@@ -0,0 +1,100 @@
+ *
+ * Neither the licensor nor any current or former employees, officers, directors, partners,
+ * trustees, representatives, agents, advisors, contractors, or volunteers of the Licensor
+ * shall be liable for any direct or indirect, special, incidental, consequential or other
+ * losses of any kind, in tort, contract or otherwise (including but not limited to loss
+ * of revenue, income or profits, or loss of use or data, or loss of reputation, or loss
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+using RadixDlt.NetworkGateway.Abstractions.Extensions;
+using System;
+
+namespace RadixDlt.NetworkGateway.Abstractions.Utilities;
+
+public static class ClampDateTimeFormatter
+{
+    // For those curious:
+    // - The beginning of this range is the start of a Gregorian calendar.
+    // - The end is simply the maximum fitting within 4 characters.
+    // - Used for compatibility with other APIs
+    private static readonly DateTimeOffset _clampMinDateTime = new(1583, 1, 1, 0, 0, 0, TimeSpan.Zero);
+    private static readonly DateTimeOffset _clampMaxDateTime = DateTimeOffset.MaxValue;
+    private static readonly long _clampMinDateTimeTimestamp = _clampMinDateTime.ToUnixTimeSeconds();
+    private static readonly long _clampMaxDateTimeTimestamp = _clampMaxDateTime.ToUnixTimeSeconds();
+
+    public static string Format(long timestamp)
```

Comment:
Can I suggest this is e.g. `FormatUnixTimestampSeconds(long unixTimestampSecs)` ?

Because this bug partly arose because we weren't clear enough between different timestamp formats on the Core API, I want to be as precise as possible ideally :D